uktrade/data-hub-frontend
# CDS-186 Adding US/Canada specific address fields to Contacts

## Description of change

This PR updates the address field to remove the forced postcode field, and adds support for dynamically showing and hiding fields based on country in order to enable state & province for company contacts.  
This work required that changes be made whereever the address field is used, since it no longer saves data to an area field but rather to two distinct fields: areaUS and areaCanada, and it is now the responsibility of the form submission to combine these according to other data in the form. This was done due to complexities involved with ensuring old data didn't persist when changing.  

## Test instructions
In Django Admin, add two features flags `address-postcode-company-required-field` and `contacts-area-field`. 
Either:  
1. Go to a US/Canadian company and add a contact.
or
2. Go to a US/Canadian contact and click edit contact.

Then:
Answer 'No' to the question 'Is the contact's address the same as the company address?'

You should be able to see an extra field _State_ for US contacts, and _Province_ for Canadian contacts. 
For US companies, there will be a _ZIP Code_ field which is mandatory. For Canadian companies, there will be a _Postal Code_ field which is mandatory.

When editing an existing contact, the previously saved area should appear in the state/province field.  

For UK contacts, the find postcode button should show and function and the state/province field should not appear.

## Screenshots
### Before
**Canadian Company**
![image](https://user-images.githubusercontent.com/70902973/132322484-81470a8c-f691-4ad9-80cb-56b4c20fb3ba.png)

**US Company**
![image](https://user-images.githubusercontent.com/70902973/132322892-23b9907a-cc5a-46e0-9040-f5968ab31786.png)

### After
**Canadian Company**
![image](https://user-images.githubusercontent.com/70902973/132326345-a87af26a-6bdc-46f3-897d-1ea7a8c11587.png)

**US Company**
![image](https://user-images.githubusercontent.com/70902973/132326102-ddd33e6f-66e4-4f9a-a8e4-df54965ab649.png)

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
